test/test_optim.py

```diff
@@ -79,28 +79,29 @@ def _test_rosenbrock_sparse(
     ):
         if scheduler_constructors is None:
             scheduler_constructors = []
-        params_t = torch.tensor([1.5, 1.5])
+        # For rosenbrock tests, it is mandated that the param is a tensor with 2 numbers
```

Comment:
nit: Maybe add an explicit `assert` to `rosenbrock()` and `drosenbrock()` to not need the 3 comments (here and before the function definitions)?
```
assert tensor.size() == torch.Size([2]), (
    f"Requires tensor with 2 scalars but got {tensor.size()}
)
```

Reply:
agreed! thanks! 

Am still going to leave the last comment just to stop anyone from experimenting with the param length